In [1]:
import numpy as np
from scipy import stats

In [2]:
np.random.seed(28)
dBP = np.random.normal(85, 10, 100000)

In [3]:
np.random.seed(12)

In [4]:
sample = np.random.choice(dBP, size=40, replace=False)

In [5]:
sample_mean = np.mean(sample)
sample_std = np.std(sample, ddof=1)
standard_error = sample_std / np.sqrt(len(sample))

In [6]:
c = stats.t.ppf(0.975, df=len(sample)-1)

In [7]:
lower_bound = sample_mean - c * standard_error
upper_bound = sample_mean + c * standard_error

In [8]:
bootstrapped_means = []

In [9]:
for i in range(1000):
    boot_sample = np.random.choice(sample, size=40, replace=True)
    bootstrapped_means.append(np.mean(boot_sample))

In [10]:
bootstrapped_means = np.array(bootstrapped_means)
bootstrapped_means.sort()

In [11]:
p2_5 = np.percentile(bootstrapped_means, 2.5)
p97_5 = np.percentile(bootstrapped_means, 97.5)

In [12]:
print("Sample mean:", sample_mean)
print("Sample standard deviation:", sample_std)
print("Standard error:", standard_error)
print("Confidence coefficient:", c)
print("Lower bound:", lower_bound)
print("Upper bound:", upper_bound)
print("2.5th percentile:", p2_5)
print("97.5th percentile:", p97_5)

Sample mean: 85.18752155840876
Sample standard deviation: 9.627818569777178
Standard error: 1.5222917789680321
Confidence coefficient: 2.0226909200367604
Lower bound: 82.10839579944351
Upper bound: 88.26664731737401
2.5th percentile: 82.20259484930493
97.5th percentile: 88.1769982523287


**Results**

Sample mean: 85.19
Sample standard deviation: 9.63
Standard error: 1.52
Confidence coefficient: 2.02
Lower bound: 82.11
Upper bound: 88.27
2.5th percentile: 82.20
97.5th percentile: 88.18

**Interpretation of Results**

The sample mean of the forty people that were chosen at random was roughly 85.19. The sample's variability is demonstrated by the sample standard deviation of 9.63. The sample mean's predicted deviation from the actual population mean was indicated by the standard error, which was 1.52.

The population mean's confidence interval was determined to be between 82.11 and 88.27 using a 95% confidence level. This indicates that the true population mean falls within this range with 95% confidence.

With the 2.5th percentile equal to 82.20 and the 97.5th percentile equal to 88.18, the bootstrapped confidence interval was likewise quite similar. The sample's consistency and dependability are supported by the bootstrap interval's near match to the t-based confidence interval.